app/Models/Beatmapset.php

```diff
@@ -1319,4 +1326,17 @@ public static function removeMetadataText($text)
 
         return preg_replace($pattern, '', $text);
     }
+
+    private static function searchCacheKey(array $params)
+    {
+        ksort($params);
+        unset($params['user']);
+
+        return 'beatmapset-search:'.serialize($params);
+    }
+
+    private static function shouldCacheSearch(array $params)
+    {
+        return !(present($params['query']) || $params['recommended']);
```

Comment:
also for `status` of 2 (favorites) and 6 (own maps) and `rank` (own ranked maps).